src/blt_mad/mad_test.go

```diff
@@ -0,0 +1,41 @@
+package blt_mad
+
+import (
+	"math"
+	"reflect"
+	"testing"
+)
+
+func TestMad(t *testing.T) {
+	// Test case 1: Empty data
+	data1 := []float64{}
+	expected1 := math.SmallestNonzeroFloat64
+	result1 := mad(data1)
+	if !reflect.DeepEqual(result1, expected1) {
```

Comment:
can you just do a straightfoward `result1 == expected1` to compare two arrays? (definitely more complex structures might need more smarts)